### PR TITLE
Post changing Hostname, BMC GUI web page need to see updated hostname

### PR DESCRIPTION
**Pull request by SteffiAntony**
Added an additional message for hostname
 - Added an additional warning message for the users regarding the Refresh of the page required after saving the hostname.
 - Defect: https://jazz07.rchland.ibm.com:13443/jazz/web/projects/CSSD#action=com.ibm.team.workitem.viewWorkItem&id=586748
Signed-off-by: Steffi Antony <“steffiantony196@gmail.com”>
image

### CODE REVIEW

File: src/views/Settings/Network/Network.vue [hunk 1]
```diff
@@ -283,7 +283,7 @@ export default {
       this.startLoader();
       this.$store
         .dispatch('network/saveHostname', modalFormData)
-        .then(this.$store.dispatch('authentication/logout'))
+        .then(() => this.$store.dispatch('authentication/logout'))
```
Comment (gtmills) [inline src/views/Settings/Network/Network.vue:286]: Why this change?

File: src/locales/en-US.json [hunk 2]
```diff
@@ -1072,7 +1072,8 @@
     "username": "Username",
     "alert": {
       "unauthorizedMessage": "User unauthorized",
-      "message": "Invalid username or password"
+      "message": "Invalid username or password",
+      "refreshMessage": "If you are failed to login, refresh the browser."
```
Comment (gtmills) [inline src/locales/en-US.json]: So any failed login we display this message..? 
The hostname is rarely changed, do we really want that edge case to influence this? We already warn the user when changing the hostname

Comment (SteffiAntony) [reply src/locales/en-US.json]: Yes.
It is rarely encountered. This happened to me last day. Then I refreshed the BMC GUI webpage and logged in with the service credentials, it worked.  Any future occurrence of this issue will trigger this toast message.

File: src/views/Settings/Network/Network.vue (hunk 1 above)
Comment (SteffiAntony) [reply src/views/Settings/Network/Network.vue:286]: Here only after the successful completion of the 'network/saveHostname' action, the 'authentication/logout' action should be dispatched. However, It is immediately invoking this.$store.dispatch('authentication/logout') instead of passing it as a callback function (Even if the hostname fails to update, the page should remain the same).

File: src/locales/en-US.json (hunk 2 above)
Comment (gtmills) [reply src/locales/en-US.json]: > "refreshMessage": "If you are failed to login, refresh the browser."

Makes it seem like a refresh will fix.. but I think for 99.9% of the time, a refresh will not help. Is this displayed any time a "Invalid username or password" is hit?

Comment (gtmills) [reply src/locales/en-US.json:1076]: I don't think we should add this message

File: src/views/Login/Login.vue [hunk 3]
```diff
@@ -172,6 +172,9 @@ export default {
       this.disableSubmitButton = true;
       const username = this.userInfo.username;
       const password = this.userInfo.password;
+      if (username === null && password === null) {
```
Comment (gtmills) [inline src/views/Login/Login.vue]: When exactly is this displayed?

File: src/locales/en-US.json (hunk 2 above)
Comment (SteffiAntony) [reply src/locales/en-US.json]: > Makes it seem like a refresh will fix.. but I think for 99.9% of the time, a refresh will not help. Is this displayed any time a "Invalid username or password" is hit?

No, It won't.

File: src/views/Login/Login.vue (hunk 3 above)
Comment (SteffiAntony) [reply src/views/Login/Login.vue]: After changing the hostname, the BMC GUI became logged out, then I tried logging back in using my service credentials, but I couldn't. It worked after I refreshed the BMC GUI webpage and logged in with my service credentials.

File: src/locales/en-US.json (hunk 2 above)
Comment (Nikhil-Ashoka) [reply src/locales/en-US.json]: It will not display if we have "Invalid username or password", But yes, we can consider not showing this message as we already show a message before making the change, And also, we would check why it is null after changing hostname.

File: src/views/Login/Login.vue (hunk 3 above)
Comment (Nikhil-Ashoka) [reply src/views/Login/Login.vue]: This is added to display this error message "If you are failed to login, refresh the browser." If we remove this message from login page, this part will also be removed.

File: src/locales/en-US.json (hunk 2 above)
Comment (SteffiAntony) [reply src/locales/en-US.json]: Removed this message as we are showing a message before making the change in hostname.

**Review by Nikhil-Ashoka — APPROVED** [review]

**Review by vedangimittal — APPROVED** [review]

**Review by nabilpandiyalayil — APPROVED** [review]

**Review by Renuka9527 — APPROVED** [review]
